src/ai/BaseAI.js

```diff
@@ -190,9 +197,13 @@ export const BaseAI = class {
             }
 
             if (this.signal?.aborted) {
-              logger.debug(`${this} Already aborted, but waiting for stream to finish`);
               done = true;
-              // Continue until stream finishes
+              if (hardStop) {
```

Comment:
pull this out / above `this.signal?.aborted` in its own thing to (1) avoid excess nesting and (2) in the unlikely case that this.signal?.aborted but hardStop is true, we still want to break